src/generators/aioseo-generator.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace Yoast\WP\Tools\Generators;
+
+use Exception;
+
+/**
+ * AIOSEO post generator.
+ *
+ * Used for testing the AIOSEO importer.
+ */
+class AIOSEO_Generator extends Core_Generator {
+
+	const META_KEYS = [
+		'_aioseo_title',
+		'_aioseo_description',
+		'_aioseo_og_title',
+		'_aioseo_og_description',
+		'_aioseo_twitter_title',
+		'_aioseo_twitter_description',
+	];
+
+	const REPLACE_VARS = [
+		'#author_first_name',
+		'#author_last_name',
+		'#author_name',
+		'#categories',
+		'#current_date',
+		'#current_day',
+		'#current_month',
+		'#current_year',
+		'#permalink',
+		'#post_content',
+		'#post_date',
+		'#post_day',
+		'#post_month',
+		'#post_title',
+		'#post_year',
+		'#post_excerpt_only',
+		'#post_excerpt',
+		'#separator_sa',
+		'#site_title',
+		'#tagline',
+		'#taxonomy_title',
+	];
+
+	/**
+	 * Generates a post with AIOSEO meta.
+	 *
+	 * @param string $post_type      The post type.
+	 * @param int[]  $author_ids     The possible author ids.
+	 * @param int[]  $attachment_ids The possible attachment ids.
+	 * @param int[]  $parent_ids     The possible parent ids.
+	 * @param int[]  $category_ids   The possible category ids.
+	 * @param int[]  $tag_ids        The possible tag ids.
+	 *
+	 * @return int The post id.
+	 *
+	 * @throws Exception If the post could not be saved.
+	 */
+	public function generate_post( $post_type, $author_ids, $attachment_ids, $parent_ids, $category_ids, $tag_ids ) {
+		$post_id = parent::generate_post( $post_type, $author_ids, $attachment_ids, $parent_ids, $category_ids, $tag_ids );
+
+		$replace_vars = \array_merge(
+			self::REPLACE_VARS,
+			$this->get_custom_field_replace_vars( $post_id )
+		);
+
+		$aioseo_fields = [
+			// "id" => 0, // Auto increment
+			"post_id"             => $post_id,
+			"title"               => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"description"         => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"og_title"            => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"og_description"      => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"twitter_title"       => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"twitter_description" => $this->generate_aioseo_replace_vars( $replace_vars ),
+			"keywords"            => null, // We don't offer keywords functionality.
+			/**
+			 * {
+			 *    "focus":{
+			 *       "keyphrase":"Test",
+			 *       "score": 0, // not important
+			 *       "analysis": "" // not important
+			 *    },
+			 *    "additional":[
+			 *       {
+			 *           "keyphrase": "Additional",
+			 *           "score": 0 // not important
+			 *           "analysis": "" // not important
+			 *       }
+			 *    ]
+			 * }
+			 */
+			"keyphrases"          => null, // JSON, see above.
+			"page_analysis"       => null, // JSON, Not important
```

Comment:
I would add a (short) comment telling why this is not important. (I guess because we can't map it to our own analysis score so it needs to be re-calculated anyway?)